src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -2896,6 +2900,31 @@ public ForkJoinTask<?> submit(Runnable task) {
 
     // Added mainly for possible use in Loom
 
+    /**
+     * Submits the given task as if submitted from a non-{@code ForkJoinTask}
+     * client. The task is added to a scheduling queue for submissions to the
+     * pool even when called from a thread in the pool.
+     *
+     * @implSpec
+     * This method is equivalent to {@link #submit(ForkJoinTask)} when called
+     * from a thread that is not in this pool.
+     *
+     * @return the task
+     * @param task the task to submit
+     * @param <T> the type of the task's result
+     * @throws NullPointerException if the task is null
+     * @throws RejectedExecutionException if the task cannot be
+     *         scheduled for execution
+     * @since 20
+     */
+    public <T> ForkJoinTask<T> externalSubmit(ForkJoinTask<T> task) {
```

Comment:
@AlanBateman Does it make sense to check the nullness of the `task` before going into the storeStoreFence etc?

Reply:
It's only to be the same as poolSubmit.